### PR TITLE
Adding Telnet 

### PR DESCRIPTION
**Pull request by Tednoob17**

### CODE REVIEW

File: docs/src/infra/protocols/telnet.md [hunk 1]
```diff
@@ -1,6 +1,144 @@
 ---
-authors: ShutdownRepo
+authors: ShutdownRepo, Tednoob17
 ---
 
 # 🛠️ Telnet
 
+## Theory
+Telnet (teletype network) is a  network protocol used to gain access to virtual terminal in local or in remote systems . He provide  bidirectional text-based communication .
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:8]: To "a" virtual terminal
"a" missing

Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:8]: "He provides" -> "It provides"

File: docs/src/infra/protocols/telnet.md [hunk 2]
```diff
@@ -1,6 +1,144 @@
...
+
+
+## Practice
+### Enumeration
+#### Banner Grabbing
+To initiate a connection with telnet server and get any information about the target .
+
+
+> [!TIP]
+> The $TARGET_PORT is optional, but the default port is 23
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:35]: Please add backticks for target port : `$TARGET_PORT`

File: docs/src/infra/protocols/telnet.md [hunk 3]
```diff
@@ -1,6 +1,144 @@
...
+
+
+#### CLI tool
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:38]: Please remove this title

File: docs/src/infra/protocols/telnet.md [hunk 4]
```diff
@@ -1,6 +1,144 @@
...
+::: tabs
+
+=== Unix-like
+
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:42]: Can you add something like: "From UNIX-like systems, multiple tools can be used to grab the telnet banner of a target."

File: docs/src/infra/protocols/telnet.md [hunk 5]
```diff
@@ -1,6 +1,144 @@
...
+```bash
+#Using Netcat
+nc -nv $TARGET_IP $TARGET_PORT
+#Using Telnet
+telnet $TARGET_IP $TARGET_PORT
+#Using Shodan-cli
+shodan stream --ports 23,1023,2323 --datadir telnet-data/ --limit 10000
+#Using nmap
+nmap -p  $TARGET_PORT -sVC  --script "*telnet* and safe" $TARGET_IP
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:51]: Can you add a newline before each comment to separate the tools?

File: docs/src/infra/protocols/telnet.md [hunk 6]
```diff
@@ -1,6 +1,144 @@
...
+```
+
+
+=== Windows
+
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:56]: Can you add: "From Windows systems [...]" just like the comment on the unix part?

File: docs/src/infra/protocols/telnet.md [hunk 7]
```diff
@@ -1,6 +1,144 @@
...
+```powershell
+nc.exe -nv $TARGET_IP $TARGET_PORT
+telnet $TARGET_IP $TARGET_PORT
+nmap -p  $TARGET_PORT   --script telnet-ntlm-info.nse $TARGET_IP
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:60]: Please add newlines between the commands, and remove the extra spaces on the last one

File: docs/src/infra/protocols/telnet.md [hunk 8]
```diff
@@ -1,6 +1,144 @@
...
+
+=== Metasploit 
+
+```
+msf > use auxiliary/scanner/telnet/telnet_version
+msf > set rhosts $TARGET_IP 
+msf > set rport $TARGET_PORT
+msf > set threads 5
+msf > exploit
+```
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:77]: I would move this block in a separate code block right under the unix commands (in the unix tab) with a sentence like "The Metasploit framework" can also be used for the job

File: docs/src/infra/protocols/telnet.md [hunk 9]
```diff
@@ -1,6 +1,144 @@
...
+:::
+
+
+### Attacks 
+
+#### Passwordless Authentication
+
+Telnet allows users to connect to a server without needing a specific identity by utilizing a passwordless login feature. This method is commonly employed for accessing or downloading public files.
+
+To connect without a password, you would use the following command:
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:87]: "You" -> "one"

File: docs/src/infra/protocols/telnet.md [hunk 10]
```diff
@@ -1,6 +1,144 @@
...
+
+```bash
+telnet $TARGET_IP
+
+#provide username
+#but not provide any password
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:93]: Can you add a space after the # ?
Also make it one sentence please : "# provide username, without password"

File: docs/src/infra/protocols/telnet.md [hunk 11]
```diff
@@ -1,6 +1,144 @@
...
+msf > set rport $TARGET_PORT
+msf > set threads 5
+msf > exploit
+```
+:::
+
+
+### Attacks 
+
+#### Passwordless Authentication
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:83]: In all titles of this page, please only keep a capital letter on the first word : "Passwordless authentication"

File: docs/src/infra/protocols/telnet.md [hunk 12]
```diff
@@ -1,6 +1,144 @@
...
+
+Telnet allows users to connect to a server without needing a specific identity by utilizing a passwordless login feature. This method is commonly employed for accessing or downloading public files.
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:85]: "Telnet allows" -> "Telnet can be configured to allow"

File: docs/src/infra/protocols/telnet.md [hunk 13]
```diff
@@ -1,6 +1,144 @@
...
+```
+#### Common Credentials
+
+If anonymous login is disabled on the Telnet server, trying common usernames and passwords like admin, administrator, root, user, or test can be a good initial step. This approach is less aggressive than attempting to guess passwords through brute force and is recommended to try first when accessing a server.
+
+```bash
+telnet $TARGET_IP
+
+#provide a common username
+#provide a common password
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:103]: Space after #

File: docs/src/infra/protocols/telnet.md [hunk 14]
```diff
@@ -1,6 +1,144 @@
...
+
+=== Hydra
+
+```bash
+hydra [-L users.txt or -l user_name] [-P pass.txt or -p password] -f [-S $TARGET_PORT] telnet://$TARGET_IP 
+
+hydra -l root -P $PATH_TO/seclists/Passwords/Default-Credentials/telnet-betterdefaultpasslist.txt $TARGET_IP telnet
+```
+
+The following commands is typed in msfconsole.
```
Comment (ShutdownRepo) [inline docs/src/infra/protocols/telnet.md:125]: Please remove this sentence, or put it right under the tab name, or else it will be in the preceding tab

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]
Thank you for the pull request
There are a few changes I think would be nice to make but after that we should be good to go!
You can make the modifications directly in your current fork, no need to create a new PR from scratch 😉

**Comment by Tednoob17** [comment]
Request Solve

File: package-lock.json
Comment (AzeTIIx) [inline package-lock.json:1]: Please don't change the project dependencies

File: package.json
Comment (AzeTIIx) [inline package.json:1]: Please don't change the project dependencies

**Review by AzeTIIx — CHANGES_REQUESTED** [review]
Hey, thanks for your changes I think the content looks good to me. However there is some weird additions to the project depencies

**Pull request closed without merging**